src/main/java/org/opentripplanner/routing/algorithm/mapping/PageCursorMapper.java

```diff
@@ -0,0 +1,137 @@
+package org.opentripplanner.routing.algorithm.mapping;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import javax.annotation.Nullable;
+import org.opentripplanner.model.plan.Itinerary;
+import org.opentripplanner.model.plan.PageCursor;
+import org.opentripplanner.transit.raptor.api.request.SearchParams;
+
+public class PageCursorMapper {
```

Comment:
This is more like a builder or factory than a Mapper, but mapper work as well - you can decide what feels most natural to you.